src/test/ui/generic-associated-types/missing-where-clause-on-trait.stderr

```diff
@@ -6,6 +6,12 @@ LL |     type Assoc<'a, 'b>;
 ...
 LL |     type Assoc<'a, 'b> = () where 'a: 'b;
    |                                       ^^ impl has extra requirement `'a: 'b`
+   |
+help: remove the `where` clause
+   |
+LL -     type Assoc<'a, 'b> = () where 'a: 'b;
+LL +     type Assoc<'a, 'b> = () ;
```

Comment:
This spacing sucks, but I have no idea how to extend a `Span` backwards while the preceding characters are spaces.

Reply:
I could have sworn there was a hacky method to do exactly that in `Session`, but there's this function that has enough logic for you to replicate what you want

https://github.com/rust-lang/rust/blob/75ff5c7dd3c50371f77ae29d43f87343d44b3829/compiler/rustc_resolve/src/diagnostics.rs#L2213

But this is a *terrible* thing. Synthesizing Spans is a sure way to arrive at madness, it's hard to maintain *and* doesn't account for macros. Even the simple `span.to(span)` creation of Spans is fraught in the face of macros :-/

All that to say, yes, you can fix this, but I wouldn't sweat it too much, proper formatting isn't the absolute goal for suggestions.

Reply:
Agree, which is why I basically didn't try too hard to do this. Users can rust rustfmt 😅